extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -548,6 +550,11 @@ public String apply(DataSegment input)
         throw e;
       }
 
+      if (!publishedSuccessfully){
```

Comment:
I think this will lead to improper behavior when a task is asked to stop for "normal" reasons (such as schema change). Maybe this is better dealt with at the supervisor level.